macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,200 @@
+    04: flush_cmd
+    10: wait_cmd
+    11: pause_cmd
+    12: resume_cmd
+    13: call_back_cmd
+    14: sync_cmd
+    15: empty_cmd
+    40: freq_duration_cmd
+    41: rest_cmd
+    42: freq_cmd
+    43: amp_cmd
+    44: timbre_cmd
+    60: wave_table_cmd
+    61: phase_cmd
+    80: sound_cmd
+    81: buffer_cmd
+    82: rate_cmd
+  sound_header_type:
+    0x00: standard
+    0xFF: extended
+    0xFE: compressed
+doc-ref: "https://developer.apple.com/library/archive/documentation/mac/pdf/Sound/Sound_Manager.pdf"
+seq:
+  - id: format
+    type: u2
+  - id: num_data_formats
+    -orig-id: number_of_data_formats
+    type: u2
+    if: format == 1
+  - id: data_formats
+    type: data_format
+    repeat: expr
+    repeat-expr: num_data_formats
+    if: format == 1
+  - id: reference_count
+    type: u2
+    if: format == 2
+  - id: num_sound_commands
+    -orig-id: number_of_sound_commands
+    type: u2
+  - id: sound_commands
+    type: sound_command
+    repeat: expr
+    repeat-expr: num_sound_commands
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    seq:
+      - id: raw_cmd
+        type: u2
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      cmd:
+        value: raw_cmd & 0x7FFF
+        enum: cmd_type
+      is_data_offset:
+        value: raw_cmd & 0x8000
+      sound_header_type:
+        pos: param2 + 20
+        type: u1
+        enum: sound_header_type
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
+      sound_header:
+        pos: param2
+        size-eos: true
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
+        type:
+          switch-on: sound_header_type
+          cases:
+            sound_header_type::standard: standard_sound_header
+            sound_header_type::extended: extended_sound_header
+            sound_header_type::compressed: compressed_sound_header
+  standard_sound_header:
+    seq:
+      - id: sample_ptr
+        type: u4
+      - id: length
+        type: u4
+      - id: sample_rate
+        type: u2
+      - id: sample_rate_frac
+        type: u2
+      - id: loop_start
+        type: u4
+      - id: loop_end
+        type: u4
+      - id: encode
+        type: u1
+        enum: sound_header_type
+      - id: base_frequency
+        type: u1
+      - id: sample_area
+        size: length
+  extended_sound_header:
+    seq:
+      - id: sample_ptr
+        type: u4
+      - id: num_channels
+        type: u4
+      - id: sample_rate
+        type: u2
+      - id: sample_rate_frac
+        type: u2
+      - id: loop_start
+        type: u4
+      - id: loop_end
+        type: u4
+      - id: encode
+        type: u1
+        enum: sound_header_type
+      - id: base_frequency
+        type: u1
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        #this is a float80
```

Comment:
I think this comment should be put into the `doc` key instead, because it's quite important thing to know when you're a programmer using this spec and you're wondering how to interpret this value. The value of `doc` is usually put into the docblock in the generated code, so a smart IDE can show this comment in a autocomplete box, which makes sense in this case.

Reply:
I agree. I do want to add some doc keys anyway. What is the policy on taking documentation bits from the original documentation? Is that allowed, or should I better rephrase?

Reply:
> What is the policy on taking documentation bits from the original documentation? Is that allowed, or should I better rephrase?

Uhh, I'm no legal advisor at all, but I think it's allowed, given that you state the original documentation in the `doc-ref` key, as you already do. Anyway, I don't recommend pasting whole sentences to the `doc` if it's not strictly necessary, not from legal reasons, but for better readability. It's generally better to keep the descriptions as concise as possible, and there are usually a lot of unnecessary words in a full sentence, whereas often a few words are enough.

Reply:
> What is the policy on taking documentation bits from the original documentation? Is that allowed, or should I better rephrase?

Strictly speaking you're not allowed to use text from Apple's documentation, because of copyright. In practice I don't think anyone will care though, as long as you make it clear where the text comes from (which is a good idea anyway).

> Anyway, I don't recommend pasting whole sentences to the `doc` if it's not strictly necessary, not from legal reasons, but for better readability. It's generally better to keep the descriptions as concise as possible, and there are usually a lot of unnecessary words in a full sentence, whereas often a few words are enough.

The original docs aren't always long sentences - for example Apple's older docs often have short descriptions like this:

```pascal
CONST
   stdSH    = $00;    {standard sound header}
   extSH    = $FF;    {extended sound header}
   cmpSH    = $FE;    {compressed sound header}
```

Reply:
I have added some documentation from the apple docs. Would that be ok? If so I would do the same for the enums, if not, I can remove it again.